Sources/TweetNacl/TweetNacl.swift

```diff
@@ -20,47 +22,64 @@ struct NaclUtil {
     }
     
     static func checkLengths(key: Data, nonce: Data) throws {
-        if key.count != crypto_secretbox_KEYBYTES {
-            throw(NaclUtilError.badKeySize)
+        if key.count != Constants.Secretbox.keyBytes {
+            throw NaclUtilError.badKeySize
         }
         
-        if nonce.count != crypto_secretbox_NONCEBYTES {
+        if nonce.count != Constants.Secretbox.nonceBytes {
             throw NaclUtilError.badNonceSize
         }
     }
     
     static func checkBoxLength(publicKey: Data, secretKey: Data) throws {
-        if publicKey.count != crypto_box_PUBLICKEYBYTES {
-            throw(NaclUtilError.badPublicKeySize)
+        if publicKey.count != Constants.Box.publicKeyBytes {
+            throw NaclUtilError.badPublicKeySize
         }
         
-        if secretKey.count != crypto_box_SECRETKEYBYTES {
-            throw(NaclUtilError.badSecretKeySize)
+        if secretKey.count != Constants.Box.secretKeyBytes{
+            throw NaclUtilError.badSecretKeySize
         }
     }
     
-    public static func randomBytes(length: Int) throws -> Data {
-        var data = Data(count: length)
-        let result = data.withUnsafeMutableBytes {
-            return SecRandomCopyBytes(kSecRandomDefault, length, $0)
+    public static func secureRandomData(count: Int) throws -> Data {
+        // Generation method is platform dependent
+        // (The Security framework is only available on Apple platforms).
+        #if os(Linux)
+
+        var bytes = [UInt8]()
+        for _ in 0..<count {
+            let randomByte = UInt8.random(in: UInt8.min...UInt8.max)
```

Comment:
https://developer.apple.com/documentation/swift/systemrandomnumbergenerator
This might help to get a cryptographically secured random on Linux.

Reply:
@anhnguyenbitmark FYI, a `SystemRandomNumberGenerator` is actually the default CSRNG for the `random(in:)` method, see the "Overview" discussion in the link you posted
> When you generate random values, shuffle a collection, or perform another operation that depends on random data, this type is the generator used by default. For example, the two method calls in this example are equivalent:
```swift
let x = Int.random(in: 1...100)
var g = SystemRandomNumberGenerator()
let y = Int.random(in: 1...100, using: &g)
```
So I believe it is correct to leave this code as-is, unless I've misunderstood the documentation or your comment?

Reply:
Ha, I didn't know it uses `SystemRandomNumberGenerator` by default. You are right.